dev/com.ibm.ws.st.liberty.gradle/src/com/ibm/ws/st/liberty/gradle/internal/LibertyGradle.java

```diff
@@ -76,57 +73,79 @@ public static LibertyBuildPluginConfiguration getLibertyGradleProjectConfigurati
-        ILaunchConfigurationType launchConfigurationType = launchManager.getLaunchConfigurationType("org.eclipse.buildship.core.launch.runconfiguration");
 
-        // What is the gradle equivalent for skipping tests?
-        // We can run specific tasks and the tests will not be run?
-        String gradleOptions = options + " -x Tests";
+		// Add the arguments to skip the tests
+		List<String> arguments = args != null ? Arrays.asList(args) : new ArrayList<String>();
+		for (String arg : LibertyGradleConstants.SKIP_TESTS_ARGS) {
+			arguments.add(arg);
+		}
+		
+		if (Trace.ENABLED) {
+			Trace.trace(Trace.INFO, "Running gradle tasks: " + Arrays.toString(tasks) + ", with arguments: " + arguments);
+		}
+
+		ProjectConnection connection = null;
         try {
-            Trace.trace(Trace.INFO, "Running gradle command with options \'" + gradleOptions + "\' with working directory: " + workingDir.toOSString());
-            ILaunchConfigurationWorkingCopy workingCopy = launchConfigurationType.newInstance((IContainer) null, workingDir.lastSegment());
-
-            workingCopy.setAttribute(IJavaLaunchConfigurationConstants.ATTR_WORKING_DIRECTORY, workingDir.toOSString());
-            workingCopy.setAttribute("tasks", gradleOptions);
-            workingCopy.setAttribute("working_dir", workingDir.toOSString());
-
-            ILaunchConfiguration launchConfig = workingCopy.doSave();
-
-            ILaunch launch = launchConfig.launch("run", monitor);
-            Trace.trace(Trace.INFO, "Running gradle command \'" + gradleOptions + "\' with working directory: " + workingDir.toOSString());
-            IProcess[] processes = launch.getProcesses();
-            List<IStreamMonitor> streamMonitors = new ArrayList<IStreamMonitor>();
-            IStreamListener streamListener = new IStreamListener() {
-                @Override
-                public void streamAppended(String text, IStreamMonitor monitor) {
-                    Trace.trace(Trace.INFO, workingDir.lastSegment() + ": " + text);
-                }
-            };
-            for (IProcess p : processes) {
-                IStreamMonitor outStream = p.getStreamsProxy().getOutputStreamMonitor();
-                outStream.addListener(streamListener);
-                streamMonitors.add(outStream);
-                IStreamMonitor errorStream = p.getStreamsProxy().getErrorStreamMonitor();
-                errorStream.addListener(streamListener);
-                streamMonitors.add(errorStream);
-            }
-            while (!launch.isTerminated() && !monitor.isCanceled()) {
-                Thread.sleep(500);
-            }
-            for (IStreamMonitor m : streamMonitors) {
-                try {
-                    m.removeListener(streamListener);
-                } catch (Exception e) {
-                    // ignore
-                }
-            }
+        	final Boolean[] result = new Boolean[1];
```

Comment:
This block of code isn't indented properly/aligned with the BuildLauncher line 106

Reply:
I think git does not show well when there are mixed tabs and spaces as it looks fine in an editor.  I changed all tabs in the file to spaces in the file.